synapse/handlers/federation_event.py

```diff
@@ -955,6 +968,18 @@ async def _compute_event_context_with_maybe_missing_prevs(
         seen = await self._store.have_events_in_timeline(prevs)
         missing_prevs = prevs - seen
 
+        # If we've already recently attempted to pull this missing event recently, don't
+        # try it again so soon. Since we have to fetch all of the prev_events, we can
+        # bail early here if we find any to ignore.
+        prevs_to_ignore = await self._store.filter_event_ids_with_pull_attempt_backoff(
+            room_id, missing_prevs
+        )
+        if len(prevs_to_ignore) > 0:
+            raise FederationPullAttemptBackoffError(
```

Comment:
This function previously raised `FederationError` on failure, so we should either make `FederationPullAttemptBackoffError` inherit from that, or make sure that we handle `FederationPullAttemptBackoffError` everywhere else.

Reply:
I decided not to inherit from `FederationError` because then we have to contend with making sure a backoff error doesn't record another failed pull attempt in the `FederationError` handlers based on the error code which is what I moved away from in https://github.com/matrix-org/synapse/pull/13816#discussion_r987676306 and https://github.com/matrix-org/synapse/pull/13816#discussion_r987685635

Good call, updated to make sure all upstream usages of `_compute_event_context_with_maybe_missing_prevs` handle `FederationPullAttemptBackoffError` ✅